### PR TITLE
Graph Interpreter

### LINKED ISSUE #23: Graph Interpreter to run graphs on engine
Create a function that runs system graphs to modify component values.

**1. Converting AttributeRef to C++ Reference**
- [x] Create component to represent user's components
- [x] Retrieve attribute of built-in components by string
- [x] Retrieve attribute of user's components
- [x] Retrieve component based on component type name
- [x] Set values of attributes safely (`std::variant` does not check types)

**2. Converting Node to a value**
- [x] Evaluate node values

### PR DESCRIPTION
**Pull request by joeltio**
Closes #23

See progress on #23.

### CODE REVIEW

File: sim/include/component/pos2DComponent.h [hunk 1]
```diff
@@ -1,12 +1,36 @@
...
 
-    Pos2DComponent(unsigned int x, unsigned int y) : x(x), y(y) {}
+    return compDef;
+};
+
+}  // namespace
+
+namespace ics::component {
+struct Pos2DComponent : public UserComponent {
+    Pos2DComponent() : UserComponent(POS2D_COMPONENT_NAME, createCompDef()) {}
```
Comment (mrzzy) [inline sim/include/component/pos2DComponent.h:33]: What happens when you import both `pos1DComponent.h` and `pos2DComponent.h` at the same time? Does `createCompDef()` do the right thing?

Comment (joeltio) [reply sim/include/component/pos2DComponent.h:33]: `createCompDef` of both components are defined in a `namespace {}` scope, this means that it doesn't export function.

File: sim/src/index/componentType.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#include <index/componentTypeIndex.h>
```
Comment (mrzzy) [inline sim/src/index/componentType.cpp:1]: Can this file be renamed `ComponentTypeIndex` for consistency.

**Review by mrzzy — COMMENTED** [review]
Generally looks Ok. Not sure if we still want to keep built in components but not a pressing issue.
when comments resolved.

File: sim/CMakeLists.txt [hunk 3]
```diff
@@ -21,19 +21,10 @@ file(MAKE_DIRECTORY ${GENERATED_INCLUDES})
...
-list(FILTER SOURCES EXCLUDE REGEX "(\.test\.cpp$|.*main\..*)")
-file(GLOB_RECURSE TEST_SOURCES "src/*.test.cpp")
-list(FILTER TEST_SOURCES EXCLUDE REGEX ".*main\..*")
-
 # define build targets
 set(TARGET_SIM ${PROJECT_NAME})
 set(TARGET_TEST "${PROJECT_NAME}-test")
 set(TARGETS ${TARGET_SIM} ${TARGET_TEST})
-# target bento simulator's executables to c++ 20
-set(CMAKE_CXX_STANDARD 20)
```
Comment (mrzzy) [inline sim/CMakeLists.txt]: I think we need this for c++20 support.

**Review by mrzzy — APPROVED** [review]
:+1: LGTM.

**Pull request merged**